javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java

```diff
@@ -76,7 +77,35 @@ protected static void generateProtocolMethodJavadoc(ClientMethod clientMethod, J
             // Response body
             IType responseBodyType;
             if (JavaSettings.getInstance().isDataPlaneClient()) {
-                responseBodyType = clientMethod.getProxyMethod().getRawResponseBodyType();
+                // special handling for paging method
+                if (clientMethod.getType() == ClientMethodType.PagingSync || clientMethod.getType() == ClientMethodType.PagingAsync || clientMethod.getType() == ClientMethodType.PagingAsyncSinglePage || clientMethod.getType() == ClientMethodType.PagingSyncSinglePage) {
+                    String itemName = clientMethod.getMethodPageDetails().getItemName();
+                    // rawResponseType has properties: 'value' and 'nextLink'
+                    IType rawResponseType = clientMethod.getProxyMethod().getRawResponseBodyType();
+                    if (!(rawResponseType instanceof ClassType)) {
+                        throw new IllegalArgumentException("clientMethod.getProxyMethod().getRawResponseBodyType() should be ClassType for paging method.");
```

Comment:
Use `IllegalStateException` as none of these is from user input and hence not argument.